// bixolon-printer/index.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Injectable } from '@angular/core';
import { Plugin, Cordova, IonicNativePlugin } from '@ionic-native/core';
/**
 * @name Bixolon Printer
 * @description
 * This plugin does something
 *
 * @usage
 * ```typescript
 * import { BixolonPrinter } from '@ionic-native/bixolon-printer';
 *
 *
 * constructor(private bixolonPrinter: BixolonPrinter) { }
 *
 * ...
 *
 *
 * this.bixolonPrinter.functionName('Hello', 123)
 *   .then((res: any) => console.log(res))
 *   .catch((error: any) => console.error(error));
 *
 * ```
 */
var BixolonPrinter = (function (_super) {
    __extends(BixolonPrinter, _super);
    function BixolonPrinter() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    BixolonPrinter.prototype.addLine = function (json) { return; };
    BixolonPrinter.prototype.addHr = function (separator) { return; };
    BixolonPrinter.prototype.printText = function (config) { return; };
    BixolonPrinter.prototype.cutPaper = function (config) { return; };
    BixolonPrinter.prototype.getStatus = function (printStatus) { return; };
    return BixolonPrinter;
}(IonicNativePlugin));
BixolonPrinter.decorators = [
    { type: Injectable },
];
/** @nocollapse */
BixolonPrinter.ctorParameters = function () { return []; };
__decorate([
    Cordova(),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", Promise)
], BixolonPrinter.prototype, "addLine", null);
__decorate([
    Cordova(),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [String]),
    __metadata("design:returntype", Promise)
], BixolonPrinter.prototype, "addHr", null);
__decorate([
    Cordova({
        callbackOrder: 'reverse'
    }),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", Promise)
], BixolonPrinter.prototype, "printText", null);
__decorate([
    Cordova({
        callbackOrder: 'reverse'
    }),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", Promise)
], BixolonPrinter.prototype, "cutPaper", null);
__decorate([
    Cordova({
        callbackOrder: 'reverse'
    }),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Boolean]),
    __metadata("design:returntype", Promise)
], BixolonPrinter.prototype, "getStatus", null);
BixolonPrinter = __decorate([
    Plugin({
        pluginName: 'BixolonPrinter',
        plugin: 'cordova-plugin-bixolon-print',
        pluginRef: 'BixolonPrint',
        repo: '',
        platforms: ['Android', 'iOS'],
    })
], BixolonPrinter);
export { BixolonPrinter };
